pkg/controller/clusterdeployment/metrics.go

```diff
@@ -61,8 +63,22 @@ var (
 			Buckets: []float64{10, 30, 60, 300, 600, 1200, 1800},
 		},
 	)
+	metricProvisionFailedTerminal = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "hive_cluster_deployments_provision_failed_terminal_total",
+		Help: "Counter incremented when a cluster provision has failed and won't be retried.",
+	},
+		[]string{"clusterpool_namespacedname"},
+	)
 )
 
+func incProvisionFailedTerminal(cd *hivev1.ClusterDeployment) {
```

Comment:
If for every CD the counter is incremented, then it should also be decremented if the relevant clusterdeployment is deleted.
I think you should make this metric a guage and report it every time for how many provisions failed.
Note that prometheus doesn't fire a change in metric until it is actually changed. So firing the metrics.Set multiple times for the same value is fine.

Reply:
> If for every CD the counter is incremented, then it should also be decremented if the relevant clusterdeployment is deleted.

We're using a counter because we want to track how many times pool CDs failed to provision, ever. We'll use prometheus to report things like the rate at which these failures are occurring.

> I think you should make this metric a guage

We [talked about](https://github.com/openshift/hive/pull/1512#discussion_r695761869) metrics when doing stale CD replacement, which is very similar to this. We [discussed](https://github.com/openshift/hive/pull/1512#discussion_r696957374) using a (separate) gauge metric to indicate the number of stale CDs in a pool at any given time, but ended up [deciding YAGNI](https://github.com/openshift/hive/pull/1512#discussion_r698782083).

Reply:
Ah! This clarifies things - I had misjudged what the metric was used for